src/game/AI/ScriptDevAI/scripts/outland/hellfire_citadel/shattered_halls/mob_shattered_hand_legionnaire.cpp

```diff
@@ -36,22 +60,82 @@ struct mob_shattered_hand_legionnaireAI : public ScriptedAI
         m_bIsRegularMode = pCreature->GetMap()->IsRegularDifficulty();
         Reset();
+
+        uint32 guid = m_creature->GetGUIDLow();
+
+        if (guid == FIRST_LEGIONNAIRE_GUID)
+            legionnaireGuid = 1;
+        else
+            legionnaireGuid = 0;
     }
 
     ScriptedInstance* m_pInstance;
     bool m_bIsRegularMode;
 
     uint32 m_uiPummelTimmer;
     uint32 m_uiAuraTimer;
+    uint32 m_uiMinionSpawnTimmer;
+    uint32 legionnaireGuid;
+    bool nearbyFriendDied;
 
     void Reset() override
     {
         m_uiPummelTimmer = urand(10000, 15000);
         m_uiAuraTimer = 15000;
+
+        nearbyFriendDied = false;
+        m_uiMinionSpawnTimmer = 0;
+    }
+
+    void ReceiveAIEvent(AIEventType eventType, Unit* pSender, Unit* pInvoker, uint32 /*miscValue*/) override
+    {
+        // Nearby creatue died
+        if (eventType == AI_EVENT_CUSTOM_EVENTAI_B)
+            nearbyFriendDied = true;
+    }
+
+    void SummonedMovementInform(Creature* pSummoned, uint32 uiMotionType, uint32 uiPointId) override
+    {
+        // When last waypoint reached, search for players.
+        if (pSummoned->GetEntry() == MOB_FEL_ORC && uiPointId == 100)
+        {
+            m_creature->CastSpell(m_creature, SPELL_ENRAGE, TRIGGERED_NONE);
+            pSummoned->GetMotionMaster()->MoveIdle();
+            pSummoned->StopMoving();
+            pSummoned->GetMotionMaster()->Clear(false, true);
+            pSummoned->SetInCombatWithZone();
+            if (Unit* pTarget = m_creature->SelectAttackingTarget(ATTACKING_TARGET_RANDOM, 0))
+                pSummoned->AI()->AttackStart(pTarget);
+        }
     }
 
     void UpdateAI(const uint32 uiDiff) override
     {
+        if (nearbyFriendDied)
+        {
+            if (!m_creature->HasAura(SPELL_ENRAGE))
+            {
+                m_creature->CastSpell(m_creature, SPELL_ENRAGE, TRIGGERED_NONE);
+                DoScriptText(EMOTE_ENRAGE, m_creature);
+            }
+
+            // Have to reasearch the cooldown for FelOrc spawn
+            if (m_uiMinionSpawnTimmer < uiDiff)
+            {                
+
+                if (Creature* felorc = m_creature->SummonCreature(MOB_FEL_ORC, FelOrcSpawnCoords[legionnaireGuid][0], FelOrcSpawnCoords[legionnaireGuid][1], FelOrcSpawnCoords[legionnaireGuid][2], FelOrcSpawnCoords[legionnaireGuid][3], TEMPSPAWN_TIMED_OOC_OR_DEAD_DESPAWN, 100000, true, true))
+                {
+                    felorc->GetMotionMaster()->MovePoint(100, FelOrcMoveCoords[legionnaireGuid][0], FelOrcMoveCoords[legionnaireGuid][1], FelOrcMoveCoords[legionnaireGuid][2]);
+                    felorc->SetRespawnCoord(FelOrcMoveCoords[legionnaireGuid][0], FelOrcMoveCoords[legionnaireGuid][1], FelOrcMoveCoords[legionnaireGuid][2], FelOrcMoveCoords[legionnaireGuid][3]);
+                }
+                DoScriptText(aRandomReinf[urand(0, 10)], m_creature);
+                nearbyFriendDied = false;
+                m_uiMinionSpawnTimmer = 5000;
```

Comment:
Its rly hard for me to find out the cooldown on FelOrc spawn after killing a legionnaire group member. 
Have to test this some more on retail.